server/api/metadata.go

```diff
@@ -1,66 +1,40 @@
 package api
 
 import (
-	"net/http"
+	"context"
 
-	"github.com/gin-gonic/gin"
-	"github.com/odpf/stencil/server/models"
+	"github.com/odpf/stencil/server/api/v1/pb"
 	"github.com/odpf/stencil/server/snapshot"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
 )
 
-// ListNames lists descriptor entries
-func (a *API) ListNames(c *gin.Context) {
-	namespace := c.Param("namespace")
-	result, err := a.Metadata.ListNames(c.Request.Context(), namespace)
+// List returns list of snapshots. If filters applied it will return filtered snapshot list
+func (a *API) List(ctx context.Context, req *pb.ListSnapshotRequest) (*pb.SnapshotList, error) {
+	res := &pb.SnapshotList{}
+	list, err := a.Metadata.List(ctx, &snapshot.Snapshot{Namespace: req.Namespace, Name: req.Name, Version: req.Version, Latest: req.Latest})
 	if err != nil {
-		c.Error(err).SetMeta(models.ErrUnknown)
-		return
+		return res, err
 	}
-	c.JSON(http.StatusOK, result)
-}
-
-// ListVersions lists version numbers for specific name
-func (a *API) ListVersions(c *gin.Context) {
-	namespace := c.Param("namespace")
-	name := c.Param("name")
-	result, err := a.Metadata.ListVersions(c.Request.Context(), namespace, name)
-	if err != nil {
-		c.Error(err).SetMeta(models.ErrUnknown)
-		return
+	for _, j := range list {
+		res.Snapshots = append(res.Snapshots, fromSnapshotToProto(j))
 	}
-	c.JSON(http.StatusOK, result)
+	return res, nil
 }
 
-//GetLatestVersion return latest version number
-func (a *API) GetLatestVersion(c *gin.Context) {
-	namespace := c.Param("namespace")
-	name := c.Param("name")
-	snapshot, err := a.Metadata.GetSnapshot(c.Request.Context(), namespace, name, "", true)
+// UpdateLatest marks specified snapshot as latest
+func (a *API) UpdateLatest(ctx context.Context, req *pb.UpdateLatestRequest) (*pb.Snapshot, error) {
```

Comment:
`UpdateLatestRequest` has a field `latest` in the proto but it's not being used anywhere right now? Do we need that field?

Reply:
Ya it's redundant, Not needed. This API just modifies latest field. All other fields on snapshot is immutable...only latest field is modifiable. 
We can move it to below format, but it won't have request body 
```
 patch /v1/snapshots/{id}/updatelatest
``` 

You can suggest, if you have something else in mind.

Reply:
Resolving this since we had related discussion at https://github.com/odpf/proton/pull/36 
For future references, the final endpoint - `patch: "/v1/snapshots/{id}:promote"`